bin/improver-nowcast-optical-flow

```diff
@@ -0,0 +1,141 @@
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Script to calculate optical flow advection velocities with option to
+extrapolate."""
+
+import datetime
+import os
+import iris
+import numpy as np
+
+from improver.argparser import ArgParser
+from improver.nowcasting.optical_flow import AdvectField, OpticalFlow
+from improver.utilities.filename import generate_file_name
+from improver.utilities.load import load_cube
+from improver.utilities.save import save_netcdf
+
+
+def main():
+    """Calculate optical flow advection velocities and (optionally)
+    extrapolate data."""
+
+    parser = ArgParser(
+        description="Calculate optical flow components from input fields "
+        "and (optionally) extrapolate to required lead times.")
+
+    parser.add_argument("input_filepaths", metavar="INPUT_FILEPATHS",
+                        nargs="+", type=str, help="Paths to the input radar "
+                        "files.  There should be 3 input files at T, T-1 and "
+                        "T-2 from which to calculate optical flow velocities.")
+    parser.add_argument("--output_dir", metavar="OUTPUT_DIR", type=str,
+                        default='', help="Directory to write output files.")
+
+    # OpticalFlow plugin configurable parameters
+    parser.add_argument("--ofc_box_size", type=int, default=30, help="Size of "
+                        "square 'box' (in grid squares) within which to solve "
+                        "the optical flow equations.")
+    parser.add_argument("--smart_smoothing_iterations", type=int, default=100,
+                        help="Number of iterations to perform in enforcing "
+                        "smoothness constraint for optical flow velocities.")
+
+    # AdvectField options
+    parser.add_argument("--extrapolate", action="store_true", default=False,
+                        help="Optional flag to advect current data forward to "
+                        "specified lead times.")
```

Comment:
Do we need the option to store the extrapolated data in a different directory to the flow velocity files and / or option to overwrite output names.